Dynatrace/dynatrace-operator
# Refactors `Reconciler` interface

# Description

Changes the `Reconciler` interface to remove the `updated` return value.
It was never according to its intention and mostly ignored by other components.
These changes remove it form the interface.

## How can this be tested?

* Create a cluster and deploy the Operator
* Deploy different kinds of Dynakubes to test a variety of features
  * Deploy an activegate statefulset
  * Deploy an agent daemonset
* Make sure the management of these resources is as expected
  * E.g., make sure no create-delete cycle was introduced due to changing behaviour

## Checklist
